src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -3775,57 +3780,43 @@ InvalidateMetadataSystemCache(void)
 
 
 /*
- * DistTableOidList iterates over the pg_dist_partition table and returns
- * a list that consists of the logicalrelids.
+ * CitusTableIdList is a wrapper around CitusTableIdListByPredFunc that
+ * returns list OID's for citus tables.
  */
 List *
-DistTableOidList(void)
+CitusTableIdList(void)
 {
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 0;
-	List *distTableOidList = NIL;
-
-	Relation pgDistPartition = table_open(DistPartitionRelationId(), AccessShareLock);
-
-	SysScanDesc scanDescriptor = systable_beginscan(pgDistPartition,
-													InvalidOid, false,
-													NULL, scanKeyCount, scanKey);
-
-	TupleDesc tupleDescriptor = RelationGetDescr(pgDistPartition);
-
-	HeapTuple heapTuple = systable_getnext(scanDescriptor);
-	while (HeapTupleIsValid(heapTuple))
-	{
-		bool isNull = false;
-		Datum relationIdDatum = heap_getattr(heapTuple,
-											 Anum_pg_dist_partition_logicalrelid,
-											 tupleDescriptor, &isNull);
-		Oid relationId = DatumGetObjectId(relationIdDatum);
-		distTableOidList = lappend_oid(distTableOidList, relationId);
-
-		heapTuple = systable_getnext(scanDescriptor);
-	}
+	return CitusTableIdListByPredFunc(NULL);
+}
 
-	systable_endscan(scanDescriptor);
-	table_close(pgDistPartition, AccessShareLock);
 
-	return distTableOidList;
+/*
+ * CitusTableIdList is a wrapper around CitusTableIdListByPredFunc that
```

Comment:
ReferenceTableIdList

Also I see mixed usages of Oid (in comments) and Ids. I would say lets go with `Oid` as we had a discussion on that and preferred Oid over id AFAIR